include/tvm/relay/attrs/algorithm.h

```diff
@@ -50,14 +50,14 @@ struct ArgsortAttrs : public tvm::AttrsNode<ArgsortAttrs> {
 };
 
 struct TopKAttrs : public tvm::AttrsNode<TopKAttrs> {
-  int k;
+  Expr k;
```

Comment:
Should we use `Optional<Integer>` here and it only holds value when input `k` is a constant?

Reply:
This is another way we can go. With this method, we don't need to fetch data from constant expr for constant attr values. However, the downside is that for symbolic value we will have null attrs. This can affect alter op layout or other operation which can only access attrs. Thus I prefer to use Expr here. We can still allow passing int or list in relay python api. Do you think this make sense?

Reply:
I think transformation pass needs to handle the cases when attrs is not constant. `attrs` are designed to only hold constant values to make shape inference easier. I think adding `Expr` to attrs is only a temporary work around. Since we now have `Optional`, we should leverage it to represent the cases that attrs are not constant.

cc @jroesch @tqchen @junrushao1994  What do you guys think about this?

Reply:
I think finally it is worthdoing to unify args and attrs, because constant attrs are only useful when dealing with simplest CNNs.

Reply:
Agree this is just a temp solution before we can unify symbolic args with attrs. Conceptually we don't need attrs to include these inputs any more, no matter they are actually constant or not.

Reply:
I agree that `Optional<Integer>` is better suited here.